src/groups/bmq/bmqimp/bmqimp_event.h

```diff
@@ -731,32 +752,20 @@ inline int Event::numCorrrelationIds() const
     BSLS_ASSERT_SAFE(d_rawEvent.isAckEvent() || d_rawEvent.isPutEvent() ||
                      d_rawEvent.isPushEvent());
 
-    return static_cast<int>(d_correlationIds.size());
+    return static_cast<int>(d_contexts.size());
 }
 
-inline const bmqt::CorrelationId& Event::correlationId(int position) const
+inline const Event::MessageContext& Event::context(int position) const
 {
     // PRECONDITIONS
     BSLS_ASSERT_SAFE(type() == EventType::e_MESSAGE);
     BSLS_ASSERT_SAFE(messageEventMode() == MessageEventMode::e_READ);
     BSLS_ASSERT_SAFE(d_rawEvent.isAckEvent() || d_rawEvent.isPutEvent() ||
                      d_rawEvent.isPushEvent());
     BSLS_ASSERT_SAFE(0 <= position);
-    BSLS_ASSERT_SAFE(static_cast<int>(d_correlationIds.size()) > position);
-
-    return d_correlationIds[position].first;
-}
-
-inline unsigned int Event::subscriptionId(int position) const
-{
-    // PRECONDITIONS
-    BSLS_ASSERT_SAFE(type() == EventType::e_MESSAGE);
-    BSLS_ASSERT_SAFE(messageEventMode() == MessageEventMode::e_READ);
-    BSLS_ASSERT_SAFE(d_rawEvent.isPushEvent());
-    BSLS_ASSERT_SAFE(0 <= position);
-    BSLS_ASSERT_SAFE(static_cast<int>(d_correlationIds.size()) > position);
+    BSLS_ASSERT_SAFE(static_cast<int>(d_contexts.size()) > position);
```

Comment:
Would prefer to reorder so the smaller value is on the left
```suggestion
    BSLS_ASSERT_SAFE(position < static_cast<int>(d_contexts.size()));
```